earn/src/pages/boost/ManageBoostPage.tsx

```diff
@@ -101,6 +146,48 @@ export default function ManageBoostPage() {
 
   const borrowerAddress = boostNftAttributes?.borrower;
 
+  const modifyData = useMemo(() => {
+    const { maxSpend, zeroForOne } = computeData(cardInfo?.borrower || undefined);
+    return ethers.utils.defaultAbiCoder.encode(
+      ['int24', 'int24', 'uint128', 'uint128', 'bool'],
+      [
+        cardInfo?.position.lower || 0,
+        cardInfo?.position.upper || 0,
+        cardInfo?.position.liquidity.toString(10) || 0,
+        maxSpend.toBigNumber(),
+        zeroForOne,
+      ]
+    ) as `0x${string}`;
+  }, [cardInfo]);
+
+  const { config: configBurn } = usePrepareContractWrite({
+    address: ALOE_II_BOOST_NFT_ADDRESS[activeChain.id],
+    abi: boostNftAbi,
+    functionName: 'modify',
+    args: [ethers.BigNumber.from(nftTokenId || 0), 2, modifyData, [true, true]],
+    chainId: activeChain.id,
+    enabled: nftTokenId !== undefined && cardInfo != null && !JSBI.equal(cardInfo?.position.liquidity, JSBI.BigInt(0)),
+  });
+  let gasLimit = configBurn.request?.gasLimit.mul(110).div(100);
+  const {
+    write: burn,
+    data: burnTxn,
+    isLoading: burnIsLoading,
+    isSuccess: burnDidSucceed,
+  } = useContractWrite({
+    ...configBurn,
+    request: {
+      ...configBurn.request,
+      gasLimit,
+    },
+  });
+
+  useEffect(() => {
+    if (burnDidSucceed && burnTxn) {
+      setPendingTxn(burnTxn);
+    }
+  }, [burnDidSucceed, burnTxn, burnIsLoading, setPendingTxn]);
```

Comment:
You don't need to use a useEffect for this. You can just pass the callback to wagmi I think?